src/execution_plan.c

```diff
@@ -1694,10 +1755,72 @@ int ExecutionPlan_ExecutionsDump(RedisModuleCtx *ctx, RedisModuleString **argv,
 	return REDISMODULE_OK;
 }
 
+int ExecutionPlan_ExecutionGet(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
+	if(argc != 2){
+		return RedisModule_WrongArity(ctx);
+	}
+
+	const char* id = RedisModule_StringPtrLen(argv[1], NULL);
+	ExecutionPlan* ep = RedisGears_GetExecution(id);
+
+	if(!ep){
+		RedisModule_ReplyWithError(ctx, "execution plan does not exist");
+		return REDISMODULE_OK;
+	}
+
+    pthread_mutex_lock(&epData.mutex);
+	RedisModule_ReplyWithArray(ctx, 10);
+	RedisModule_ReplyWithStringBuffer(ctx, "status", strlen("status"));
+	if(ep->isDone){
+		RedisModule_ReplyWithStringBuffer(ctx, "done", strlen("done"));
+	}else{
+		RedisModule_ReplyWithStringBuffer(ctx, "running", strlen("running"));
+	}
+
+	RedisModule_ReplyWithStringBuffer(ctx, "state", strlen("state"));
+	if(ep->isErrorOccure){
+		RedisModule_ReplyWithStringBuffer(ctx, "error", strlen("error"));
+		RedisModule_ReplyWithStringBuffer(ctx, "message", strlen("message"));
+        // TODO: replace with a full backtrace
+        assert(RedisGears_GetRecordsLen(ep) > 0);
+		Record* record = RedisGears_GetRecord(ep, 0);
+        assert(RedisGears_RecordGetType(record) == ERROR_RECORD);
```

Comment:
the asserts here are wrong, but I guess you fix it when you add the error records list.

Reply:
Yep - this is just a placeholder.

Reply:
This should be resolved via a97de6e